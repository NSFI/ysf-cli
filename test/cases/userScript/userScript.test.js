const path = require("path");
const fsExtra = require("fs-extra");

describe("UpdateMethod", () => {
  beforeAll(() => {
    process.chdir(__dirname);
    jest.mock("../../../src/update/variables", () => {
      const path = require("path");
      return {
        PROJ_DIR: path.resolve("./"), //根目录
        CACHE_DIR: path.resolve("./")
      };
    });
  });

  it("should execute custom script correctly", async () => {
    //  prepare
    const updateMethod = require("../../../src/update/updateMethods");

    const updateConfig = await fsExtra.readJSON(
      path.resolve(__dirname, "./u/update.json")
    );


    //  execute
    let output = await updateMethod.execUserScript(updateConfig);


    //  verify
    expect(output).toMatchSnapshot();
  });
  afterAll(() => {
    jest.resetAllMocks();
  });
});
