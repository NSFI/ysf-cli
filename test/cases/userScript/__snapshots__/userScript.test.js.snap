// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateMethod should execute custom script correctly 1`] = `
Array [
  "===========BEGIN==============
const React = require(\\"React\\");

class TEST extends React.Component {
  render() {
    return <h2>test case </h2>;
  }
}
module.exports = TEST;

===========END==============",
  "===========BEGIN==============
firstLetter
HELLOWORLD
BIGBANG@@@
===========END==============",
  "===========BEGIN==============
process.argv.length =  0
===========END==============",
  "===========BEGIN==============
Special FOlders\\"$1
|       FOR NODEJS
 \\\\-----  sPecila File.txt

===========END==============",
  "Special FOlders\\"$1
|     FOR BASH
 \\\\-----  sPecila File.txt",
]
`;
